gradebookng/tool/src/webapp/scripts/gradebook-update-ungraded.js

```diff
@@ -1,80 +1,144 @@
+      return null;
+    }
+
+    // Use TrimPath to process the template
+    const templateHtml = templateElement.innerHTML.trim();
+    const modalTemplate = TrimPath.parseTemplate(templateHtml);
+    const processedHtml = modalTemplate.process({
+      score,
+      group,
+      isExtraCredit
+    });
+
+    // Create the modal element
+    const modalElement = document.createElement('div');
+    modalElement.innerHTML = processedHtml;
+    
+    // Ensure proper z-index stacking for nested modals
+    const firstModal = document.querySelector('.modal');
+    if (firstModal) {
+      const currentZIndex = parseInt(window.getComputedStyle(firstModal).zIndex, 10);
+      modalElement.style.zIndex = (currentZIndex + 10).toString();
+    }
+
+    return modalElement.firstElementChild;
+  }
 
-  function handleFakeSubmit(event) {
-    event.preventDefault();
-    event.stopPropagation();
+  showConfirmation() {
+    const modalElement = this.buildConfirmationModal();
+    if (!modalElement) return;
+    
+    document.body.appendChild(modalElement);
+
+    // Configure Bootstrap modal with nested modal options
+    const modal = new bootstrap.Modal(modalElement, {
+      backdrop: 'static', // Prevent clicking outside to close when nested
+      keyboard: true // Allow escape key
+    });
+    
+    // Store reference to parent modal
+    const parentModal = this.content.closest('.wicket-modal');
+    const parentBootstrapModal = parentModal ? bootstrap.Modal.getInstance(parentModal) : null;
+    
+    modalElement.addEventListener('hidden.bs.modal', () => {
+      modalElement.remove();
+      // Return focus to the group selector in parent modal
+      this.content.querySelector('.gb-update-ungraded-group').focus();
+    });
+
+    modalElement.addEventListener('show.bs.modal', () => {
+      if (parentModal) {
+        // Adjust margin to position relative to parent modal
+        modalElement.style.marginTop = `${parentModal.offsetTop + 40}px`;
+      }
+    });
+
+    modalElement.addEventListener('shown.bs.modal', () => {
+      // Set focus to cancel button
+      modalElement.querySelector('.gb-update-ungraded-cancel').focus();
+    });
+
+    // Handle escape key manually to ensure proper modal stacking
+    modalElement.addEventListener('keydown', (event) => {
+      if (event.key === 'Escape') {
+        event.stopPropagation(); // Prevent parent modal from catching escape
+        modal.hide();
+      }
+    });
+
+    modalElement.querySelector('.gb-update-ungraded-continue').addEventListener('click', () => {
+      this.performRealSubmit();
+      modal.hide();
+    });
+
+    modal.show();
+  }
 
-    showConfirmation();
+  performRealSubmit() {
+    const realSubmitButton = this.content.querySelector('.gb-update-ungraded-real-submit');
+    realSubmitButton.click();
+    this.content.querySelectorAll('input, select, button').forEach(input => {
+      input.disabled = true;
+    });
```

Comment:
do you think, adding a style class `input-disabled` would be a good option here?